editor/src/components/inspector/sections/comment-section.tsx

```diff
@@ -70,61 +78,46 @@ const ThreadPreviews = React.memo(() => {
     setOpen((prevOpen) => !prevOpen)
   }, [setOpen])
 
-  const { threads: threads } = useThreads()
-  const { threads: resolvedThreads } = useResolvedThreads()
+  const { threads } = useThreads()
   const { threads: readThreads } = useReadThreads()
 
   sortThreadsByDescendingUpdateTimeInPlace(threads)
-  sortThreadsByDescendingUpdateTimeInPlace(resolvedThreads)
-  sortThreadsByDescendingUpdateTimeInPlace(readThreads)
 
-  const showResolved = useEditorState(
+  const commentFilterMode = useEditorState(
     Substores.restOfEditor,
-    (store) => store.editor.showResolvedThreads,
-    'ThreadPreviews showResolvedThreads',
+    (store) => store.editor.commentFilterMode,
+    'ThreadPreviews commentFilterMode',
   )
 
-  const toggleShowResolved = React.useCallback(() => {
-    dispatch([
-      setShowResolvedThreads(!showResolved),
-      switchEditorMode(EditorModes.selectMode(null, false, 'none')),
-    ])
-  }, [showResolved, dispatch])
-
-  const [sortByDateNewestFirst, setSortByDateNewestFirst] = React.useState(true)
-  const [sortByUnreadFirst, setSortedByUnreadFirst] = React.useState(false)
-
-  const toggleSortByDateNewestFirst = React.useCallback(() => {
-    setSortByDateNewestFirst((prevValue) => !prevValue)
-  }, [])
-  const toggleSortByUnreadFirst = React.useCallback(() => {
-    setSortedByUnreadFirst((prevValue) => !prevValue)
-  }, [])
+  const filter = React.useMemo(
+    () =>
+      filterOptions.find((option) => option.value === commentFilterMode) ?? {
+        label: 'All',
+        value: 'all',
+      },
+    [commentFilterMode],
+  )
 
   const sortedThreads = React.useMemo(() => {
-    let filteredThreads = threads
-
-    if (!showResolved) {
-      filteredThreads = filteredThreads.filter((thread) => !resolvedThreads.includes(thread))
-    }
-    if (sortByDateNewestFirst) {
-      filteredThreads = filteredThreads.slice().reverse()
-    }
-    if (sortByUnreadFirst) {
-      const unreadThreads = filteredThreads.filter((thread) => !readThreads.includes(thread))
-      const theReadThreads = filteredThreads.filter((thread) => readThreads.includes(thread))
-      filteredThreads = [...unreadThreads, ...theReadThreads]
+    switch (commentFilterMode) {
+      case 'all':
+        return threads.filter((t) => !t.metadata.resolved)
+      case 'all-including-resolved':
+        return threads
+      case 'unread-only':
+        const readThreadIds = pluck(readThreads, 'id')
+        return threads.filter((t) => !readThreadIds.includes(t.id))
```

Comment:
It seems this will still show resolved comments if they are unread

Reply:
True, fixed in 3fe97d7aa3bbdf382f0d0ad11ad190c2d92a8800